dali/kernels/context.h

```diff
@@ -31,30 +33,48 @@ struct Context<ComputeGPU> {
   cudaStream_t stream = 0;
 };
 
-class ScratchpadAllocator {
+/// @brief Interface for kernels to obtain auxiliary working memory
```

Comment:
`-Wpedantic`
Doesn't our documentation follow
```
  /**
   *
   */
```
format?
e.g. in `tensor_list.h`
```
  /**
   * @brief Returns a typed pointer to the tensor with the given index.
   */
  template <typename T>
  DLL_PUBLIC inline T* mutable_tensor(int idx) {
    return this->template mutable_data<T>() + tensor_offset(idx);
  }
```